events/tests/test_api.py

```diff
@@ -1059,6 +1062,44 @@ def test_cannot_unenrol_from_occurrence_in_past(
     assert_match_error_code(executed, PAST_ENROLMENT_ERROR)
 
 
+@freeze_time("2024-01-01 12:00:00")
+def test_cannot_unenrol_within_48_hours(guardian_api_client, child_with_user_guardian):
+    # Create occurrence 47 hours in future (within 48 hour limit)
+    future_time = timezone.now() + timedelta(hours=47)
+    enrolment = EnrolmentFactory(
+        occurrence__time=future_time, child=child_with_user_guardian
+    )
+
+    variables = deepcopy(UNENROL_OCCURRENCE_VARIABLES)
+    variables["input"]["occurrenceId"] = get_global_id(enrolment.occurrence)
+    variables["input"]["childId"] = get_global_id(child_with_user_guardian)
+
+    executed = guardian_api_client.execute(
+        UNENROL_OCCURRENCE_MUTATION, variables=variables
+    )
+
+    assert_match_error_code(executed, "TOO_LATE_TO_UNENROL_ERROR")
+
+
+@freeze_time("2024-01-01 12:00:00")
+def test_can_unenrol_outside_48_hours(guardian_api_client, child_with_user_guardian):
+    # Create occurrence 49 hours in future (outside 48 hour limit)
```

Comment:
How about setting the explicit value for the environment variable here so the test is robust against local .env file changes?
```suggestion
    settings.KUKKUU_ENROLMENT_UNENROL_HOURS_BEFORE = 48
    # Create occurrence 49 hours in future (outside 48 hour limit)
```